core/src/main/java/org/axonframework/queryhandling/SimpleQueryBus.java

```diff
@@ -188,6 +217,120 @@ public SimpleQueryBus(MessageMonitor<? super QueryMessage<?, ?>> messageMonitor,
+                               interceptedQuery.getQueryName(),
+                               interceptedQuery.getResponseType(),
+                               interceptedQuery.getUpdateResponseType()));
+            }
+            Iterator<QuerySubscription> subsIterator = subs.iterator();
+            boolean invocationSuccess = false;
+            while (!invocationSuccess && subsIterator.hasNext()) {
+                try {
+                    QuerySubscription subscription = subsIterator.next();
+                    DefaultUnitOfWork<QueryMessage<Q, I>> uow = DefaultUnitOfWork.startAndGet(interceptedQuery);
+                    if (subscription instanceof SubscribableQuerySubscription) {
+                        if (interceptedQuery.getUpdateResponseType()
+                                            .matches(((SubscribableQuerySubscription) subscription).getUpdateType())) {
+
+                            registration = invokeSubscriptionQueryHandler(uow,
+                                                                          ((SubscribableQuerySubscription) subscription)
+                                                                                  .getSubscriptionQueryHandler(),
+                                                                          updateHandler);
+                            invocationSuccess = true;
+                        }
+                    } else {
+                        invokeRegularQueryHandler(uow, subscription.getQueryHandler(), updateHandler);
+                        invocationSuccess = true;
+                    }
+                } catch (NoHandlerForQueryException e) {
+                    // Ignore this Query Handler, as we may have another one which is suitable
+                }
+            }
+            if (!invocationSuccess) {
+                throw new NoHandlerForQueryException(
+                        format("No suitable handler was found for %s with response type %s and update type %s",
+                               interceptedQuery.getQueryName(),
+                               interceptedQuery.getResponseType(),
+                               interceptedQuery.getUpdateResponseType()));
+            }
+        } catch (Exception e) {
+            monitorCallback.reportFailure(e);
+            updateHandler.onError(e);
+        }
+
+        return registration;
+    }
+
+    /**
+     * Invokes regular query handler and completes the update handler right afterwards.
+     *
+     * @param uow           the Unit of Work in which the query handler will be invoked
+     * @param queryHandler  the query handler to be invoked
+     * @param updateHandler the update handler to be invoked with result of query handler
+     * @param <Q>           the query type
+     * @param <I>           the initial result type
+     * @throws Exception propagated from query handler
+     */
+    @SuppressWarnings("unchecked")
```

Comment:
What line it's warning is being suppressed?

Reply:
Ooops, we are clear here, no need for suppressing :)